client/js/jayme.js

```diff
@@ -1,194 +1,192 @@
     }
-
-    // LISTENERS ####################################################
-    function onKeyDown(e) {
-        console.log(e.key);
-        switch(e.key) {
-            case 'e':
-                increaseIncome();
-                break;
-            case 'a':
-                setArmyStance('aggressive');
-                break;
-            case 's':
-                setArmyStance('passive');
-                break;
-            case 'r':
-                increaseArmy();
-                break;
-        }
+});
+
+socket.on('matched', function(usernameMap) {
+    clientState.oppPresent = true;
+    document.getElementById('oppUsername').innerHTML = 'Opponent: Found!';
+});
+
+function setUsername() {
+    var username = prompt("Enter your username");
+    if (username != null){
+        document.getElementById('username').innerHTML = 'You:' + username;
+        socket.emit('username', username);
     }
-
-    function prepareEventHandlers() {
-        var canvasPosition = {
-            x: canvas.offsetLeft,
-            y: canvas.offsetTop
-        };
-
-        window.addEventListener('keypress', onKeyDown, false);
-        usernameElement.addEventListener('click', setUsername, false);
-        readyButtonElement.addEventListener('click', toggleReady, false);
-
-
-        // canvas.addEventListener('click', function(e) {
-        //     var mouse = {
-        //         x: e.pageX - canvasPosition.x,
-        //         y: e.pageY - canvasPosition.y
-        //     };
-        // }, false);
-
-        // canvas.addEventListener('mousemove', function(e) {
-        //     var mouse = {
-        //         x: e.pageX - canvasPosition.x,
-        //         y: e.pageY - canvasPosition.y
-        //     };
-        // }, false);
+}
+
+function toggleReady() {
+    if (clientState.myUsername != null && clientState.myUsername != '') {
+        clientState.ready = !clientState.ready;
+        document.getElementById('playerStatus').innerHTML = clientState.ready  ? "Your Status: Ready" : "Your Status: Not Ready"
+        socket.emit('ready', clientState.ready);
+    } else {
+        setUsername();
+    }
+}
+
+// LISTENERS ####################################################
+function onKeyDown(e) {
+    console.log(e.key);
+    switch(e.key) {
+        case 'e':
+            increaseIncome();
+            break;
+        case 'a':
+            setArmyStance('aggressive');
+            break;
+        case 's':
+            setArmyStance('passive');
+            break;
+        case 'r':
+            increaseArmy();
+            break;
     }
+}
+
+function handleCanvasClick(e) {
+    var x = e.clientX + document.body.scrollLeft + document.documentElement.scrollLeft;
+    var y = e.clientY + document.body.scrollTop + document.documentElement.scrollTop;
+    x -= e.target.offsetLeft;
+    y -= e.target.offsetTop;
+
+    if ((Math.pow(200-x,2) + Math.pow(150-y,2)) < Math.pow(Math.log(clientState.income+1)*10 + 30,2)) {
```

Comment:
Had to stare at it for a while to realize this is calculating the distance between the mouse click and the center of the circle and checking that it's less than the radius.
I would start thinking about how to abstract this logic.  I'm thinking naively you have a list of components and you pass the coordinates to each "interactable" UI object and it tells you whether it has been clicked.

Another minor issue here is that you're hardcoding the circle size logic here and in the painter.